swiftide/tests/lancedb.rs

```diff
@@ -87,3 +92,125 @@ async fn test_lancedb() {
+    std::fs::write(&codefile, code).unwrap();
+
+    // Setup mock servers to simulate API responses
+    let mock_server = MockServer::start().await;
+    mock_chat_completions(&mock_server).await;
+
+    let openai_client = openai_client(&mock_server.uri(), "text-embedding-3-small", "gpt-4o");
+
+    let fastembed = FastEmbed::try_default().unwrap();
+
+    let lancedb = LanceDB::builder()
+        .uri(tempdir.child("lancedb").to_str().unwrap())
+        .vector_size(384)
+        .with_vector(EmbeddedField::Combined)
+        .with_metadata(METADATA_QA_CODE_NAME)
+        .with_metadata("filter")
+        .with_metadata("path")
+        .table_name("swiftide_test")
+        .build()
+        .unwrap();
+
+    Pipeline::from_loader(loaders::FileLoader::new(tempdir.path()).with_extensions(&["rs"]))
+        .then_chunk(ChunkCode::try_for_language("rust").unwrap())
+        .then(MetadataQACode::new(openai_client.clone()))
+        .then(|mut node: indexing::Node| {
+            // Add path to metadata, by default, storage will store all metadata fields
+            node.metadata
+                .insert("path", node.path.display().to_string());
+            node.metadata
+                .insert("filter".to_string(), "true".to_string());
+            Ok(node)
+        })
+        .then_in_batch(transformers::Embed::new(fastembed.clone()).with_batch_size(20))
+        .log_nodes()
+        .then_store_with(lancedb.clone())
+        .run()
+        .await
+        .unwrap();
+
+    let db_connection = lancedb
+        .get_connection()
+        .await
+        .expect("Failed to establish connection to LanceDB pool");
+
+    let vector_table = db_connection
+        .open_table("swiftide_test")
+        .execute()
+        .await
+        .expect("Failed to open vector search table 'swiftide_test'");
+
+    let similarity_config =
+        swift_query_pipeline::search_strategies::SimilaritySingleEmbedding::from_filter(
+            "filter = \"true\"".to_string(),
+        );
+
+    // Create the custom query strategy for vector similarity search
+    let vector_search_strategy =
+        swift_query_pipeline::search_strategies::CustomStrategy::from_query(
+            move |query_node| -> Result<lance_query_builder::VectorQuery> {
+                let embedding = query_node
+                    .embedding
+                    .as_ref()
+                    .ok_or_else(|| anyhow::anyhow!("Query embedding not found"))?;
+
+                let vector_field =
+                    lance_integration::VectorConfig::from(EmbeddedField::Combined).field_name();
+
+                let mut query_builder = vector_table
+                    .query()
+                    .nearest_to(embedding.as_slice())
+                    .context("Failed to create vector similarity query")?
+                    .column(&vector_field);
+
+                let result_limit = usize::try_from(similarity_config.top_k())
```

Comment:
Maybe just use an arbitrary number? Feels a bit complicated to just create a new search strategy, if you really need a number.